src/main/java/org/springframework/integration/kafka/dsl/KafkaMessageListenerContainerSpec.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.kafka.dsl;
+
+import java.util.regex.Pattern;
+
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.OffsetCommitCallback;
+
+import org.springframework.core.task.AsyncListenableTaskExecutor;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer;
+import org.springframework.kafka.listener.AcknowledgingMessageListener;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.listener.ErrorHandler;
+import org.springframework.kafka.listener.config.ContainerProperties;
+import org.springframework.kafka.support.TopicPartitionInitialOffset;
+
+/**
+ * A helper class in the Builder pattern style to delegate options to the
+ * {@link ConcurrentMessageListenerContainer}.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Artem Bilan
+ * @author Gary Russell
+ *
+ * @since 3.0
+ */
+public class KafkaMessageListenerContainerSpec<K, V> {
```

Comment:
Let's make it as `IntegrationComponentSpec` and add `Kafka.container()` factory method!

With an `id()` in the `IntegrationComponentSpec` I have some idea for the https://jira.spring.io/browse/INT-4252 like making `getComponentsToRegister()` returning `Map<Object, String>` to register intermediate beans under provided bean names.
WDYT?